### PR TITLE
Fixed some clippy warnings in components

### PR DESCRIPTION
**Pull request by komutumu**
This PR includes fixes to some clippy warnings:

- Elides timelines
- Removes useless use of format!
- Removes unneeded return statement
- Eliminates use of an explicit closure for copying elements.
- Replaces the use of negated comparison with ```partial_cmp```
---
- [X] `./mach build -d` does not report any errors
- [X] `./mach test-tidy` does not report any errors
- [X] These changes are part of #31500 
- [X] These changes do not require tests because it touches not functionality

### CODE REVIEW

**Comment by jahielkomu** [comment]
Hello @mrobinson , I hope you are well.
Please kindly review.

File: components/script/dom/webgl2renderingcontext.rs [hunk 1]
```diff
@@ -2953,7 +2953,10 @@ impl WebGL2RenderingContextMethods for WebGL2RenderingContext {
         }
 
         if pbo_offset < 0 || pbo_offset as usize > pixel_unpack_buffer.capacity() {
-            return Ok(self.base.webgl_error(InvalidValue));
+            return {
+                self.base.webgl_error(InvalidValue);
+                Ok(())
+            };
```
Comment (mrobinson) [inline components/script/dom/webgl2renderingcontext.rs:2959]:
```suggestion
            self.base.webgl_error(InvalidValue);
            return Ok(());
```

**Review by mrobinson — CHANGES_REQUESTED** [review]

**Comment by jahielkomu** [comment]
@mrobinson , I have updated the PR.
Kindly review.

**Review by mrobinson — APPROVED** [review]

**Comment by mrobinson** [comment]
Thanks!

**Pull request merged**